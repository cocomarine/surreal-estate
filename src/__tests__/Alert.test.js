import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(<Alert message="" />);
    expect(rendered).toMatchSnapshot();
  });

  it("displays an error message", () => {
    const { asFragment } = render(<Alert message="Error!" />);
    const errorMsg = screen.getByText(/Error/);

    expect(asFragment()).toMatchSnapshot();
    expect(errorMsg).toBeInTheDocument();
  });

  it("displays a success message", () => {
    const { asFragment } = render(<Alert message="Success!!" success />);
    const successMsg = screen.getByText(/Success/);

    expect(asFragment()).toMatchSnapshot();
    expect(successMsg).toBeInTheDocument();
  });
});
