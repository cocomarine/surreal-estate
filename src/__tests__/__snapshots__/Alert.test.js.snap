// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert renders correctly 1`] = `null`;
